File: 00001_test.ex
defmodule AnswerSeeker do
  def search_for_the_answer() do
    case :io.fread('', '~d') do
      {:ok, [42]} ->
        :found

      {:ok, [number]} ->
        IO.puts(number)
        search_for_the_answer()

      :eof ->
        :not_found
    end
  end
end

AnswerSeeker.search_for_the_answer()

File: 00001_test.erl
-module(prog).
-export([main/0]).

main() ->
  search_for_the_answer().

search_for_the_answer() ->
  case io:fread("", "~d") of
    {ok, [42]} ->
      found;

    {ok, [Number]} ->
      io:fwrite("~B~n", [Number]),
      search_for_the_answer();

    eof ->
      not_found
  end.
